service/src/main/java/org/broadinstitute/listener/relay/http/RelayedHttpRequestProcessor.java

```diff
@@ -83,6 +97,38 @@ public Result writeNotAcceptedResponseOnCaller(RelayedHttpListenerContext contex
     return Result.FAILURE;
   }
 
+  public Result writePreflightResponse(RelayedHttpListenerContext context) {
+    if (context.getResponse() == null) {
+      logger.error("The context did not have a valid response");
+      return Result.FAILURE;
+    }
+
+    RelayedHttpListenerResponse listenerResponse = context.getResponse();
+    listenerResponse.setStatusCode(204);
+    listenerResponse
+        .getHeaders()
+        .put(ACCESS_CONTROL_ALLOW_METHODS, corsSupportProperties.preflightMethods());
+
+    if (!listenerResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN))
```

Comment:
I don't think this check has any meaning/purpose. We are crafting the response, so the header won't be set in any case.

Reply:
if you're certain `context.getResponse()` will have empty header, I'm happy to remove the check